pkg/check/gc/reserve.go

```diff
@@ -178,6 +183,43 @@ func CheckReserve(c *bee.Cluster, o Options) error {
 		return fmt.Errorf("high value chunks were gc'd. Retrieved: %d,  gc'd count: %d", hasCount, len(highValueChunks)-hasCount)
 	}
 
+	pinned, err := client.GetPins(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to get pins: %w", err)
+	}
+	if len(pinned) != 1 {
+		return fmt.Errorf("unexpected pin count %d", len(pinned))
+	}
+	pinnedAddr := pinned[0]
+
+	if !pinnedChunk.Address().Equal(pinnedAddr) {
+		return fmt.Errorf("chunk %q is not pinned", pinnedChunk.Address())
+	}
+
+	if have, err := client.GetPinnedRootHash(ctx, pinnedAddr); err != nil {
+		return fmt.Errorf("unable to get pinned root hash: %w", err)
+	} else if !have.Equal(pinnedAddr) {
+		return fmt.Errorf("address mismatch: have %q; want %q", have, pinnedAddr)
+	}
+
+	if err := client.UnpinRootHash(ctx, pinnedAddr); err != nil {
+		return fmt.Errorf("cannot unpin chunk: %w", err)
+	}
+
+	if have, err := client.GetPinnedRootHash(ctx, pinnedAddr); err != nil {
+		return fmt.Errorf("unable to get pinned root hash: %w", err)
+	} else if !have.Equal(swarm.ZeroAddress) {
+		return fmt.Errorf("address mismatch: have %q; want none", have)
+	}
+
+	pinned, err = client.GetPins(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to get pins: %w", err)
+	}
+	if len(pinned) > 0 {
+		return errors.New("pin count is greater than zero")
+	}
```

Comment:
r: a potential future improvement would be to check that the chunk is actually GCd after you unpin it, but it might need a bit more changes in the test scenario to accommodate for that. this is good as a first iteration :+1:

Reply:
Agree with this, In the next iteration, I would even consider factoring out the local pinning test to its own separate check.